src/components/dialog/Dialog.jsx

```diff
@@ -97,6 +100,7 @@ function BaseDialog({
 
     if (!supportsTransitions()) {
       fireTransitionEndCallbacks();
+      setHasFinishedTransition(true);
```

Comment:
Why not to put `setHasFinishedTransition(true);` in `fireTransitionEndCallbacks`?